crates/db_schema/src/impls/images.rs

```diff
@@ -39,31 +45,40 @@ impl LocalImage {
 }
 
 impl RemoteImage {
-  pub async fn create(pool: &mut DbPool<'_>, links: Vec<Url>) -> Result<usize, Error> {
+  pub async fn create(pool: &mut DbPool<'_>, link_: &Url) -> Result<usize, Error> {
     let conn = &mut get_conn(pool).await?;
-    let forms = links
-      .into_iter()
-      .map(|url| RemoteImageForm { link: url.into() })
-      .collect::<Vec<_>>();
+    let form = RemoteImageForm {
+      link: link_.clone().into(),
+    };
     insert_into(remote_image::table)
-      .values(forms)
+      .values(form)
       .on_conflict_do_nothing()
       .execute(conn)
       .await
   }
 
   pub async fn validate(pool: &mut DbPool<'_>, link_: DbUrl) -> Result<(), Error> {
     let conn = &mut get_conn(pool).await?;
-
-    let exists = select(exists(
-      remote_image::table.filter(remote_image::link.eq(link_)),
-    ))
-    .get_result::<bool>(conn)
-    .await?;
-    if exists {
+    let res = remote_image::table
+      .find(link_)
+      .first::<RemoteImage>(conn)
+      .await
+      .optional()?;
+    if res.is_some() {
       Ok(())
     } else {
       Err(NotFound)
     }
   }
 }
+
+impl ImageDetails {
+  pub async fn create(pool: &mut DbPool<'_>, form: &ImageDetailsForm) -> Result<Self, Error> {
+    let conn = &mut get_conn(pool).await?;
+    insert_into(image_details::table)
+      .values(form)
+      .on_conflict_do_nothing()
+      .get_result::<Self>(conn)
+      .await
+  }
+}
```

Comment:
I would make this private and call it from `RemoteImage;:create()` in a transaction. So its not possible to create details without corresponding image.

Reply:
Note that with this transaction now, one of the unit tests fails, since getting the details for remote images requires a locally running pictrs server.